### ISSUE TITLE
[Bug] Tutorial Example for Single Fracture Under Shear Compression Cannot Match Analytical Solution with Parallel Run

### ISSUE THREAD

**Issue by jhuang2601** [GEOSX/GEOSX#1984]
**Describe the bug**
Reran [tutorial example for single fracture under shear compression](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/advancedExamples/validationStudies/singleFracCompression/Example.html) with latest develop repo.
It can be running, however cannot produce the expected results and deviates away from the corresponding analytical solution. 
The case was running correctly and matched well with analytical solution previously and something must be broken.

**To Reproduce**
The xml file for this tutorial example is located at: `inputFiles/lagrangianContactMechanics/ContactMechanics_SingleFracCompression_benchmark.xml`

Run it with the most recent develop repo to reproduce the reported issue:
`srun -N 1 -n 4 -p pecan /data/gpfs/Users/j0436735/travis-deployments/CPU/GEOSX-75c9fb7/bin/geosx -i ContactMechanics_SingleFracCompression_benchmark.xml -x 2 -y 2 -z 1`

and a python script can be used to postprocess the result: 
`docs/sphinx/advancedExamples/validationStudies/singleFracCompression/singleFracCompressionFigure.py`

**Screenshots**
Below is the current simulation results and obvious difference between GEOSX results and analytical solution can be observed:
![image](https://user-images.githubusercontent.com/53012159/180476584-afbea81b-d38d-4491-ab6e-bb930dcbc1cc.png)

**Platform (please complete the following information):**
- Machine: Pecan 
- GEOSX Version: develop `75c9fb7`

**Comment by jhuang2601** [comment]
@CusiniM This might be the same issue related to our tests with inclined fracture for poro-EDFM?

**Comment by CusiniM** [comment]
there s no smoke test covering this case?

**Comment by jhuang2601** [comment]
@CusiniM There is a smoke test and it should be covered in the integratedTests.

**Comment by CusiniM** [comment]
There is one. https://github.com/GEOSX/GEOSX/blob/develop/inputFiles/lagrangianContactMechanics/ContactMechanics_SingleFracCompression_smoke.xml Can you start by checking that one? 

This looks more like a indexing issue to me.

**Comment by jhuang2601** [comment]
@CusiniM You're right. with a serial run, this issue is gone
![image](https://user-images.githubusercontent.com/53012159/180482885-26496239-d1f3-4351-8a04-0a0557d90db1.png)

**Comment by PorousMedia** [comment]
Hi. I have a similar issue. I ran basic example for CO2 injection [](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/basicExamples/co2Injection/Example.html), works fine. But when I parallelized it, it gave a very different picture. I also repeatedly got an error message while it was running. 
[0ced17b6490e:00011] Read -1, expected 4264, errno = 1
[0ced17b6490e:00011] Read -1, expected 4264, errno = 1
[0ced17b6490e:00011] Read -1, expected 4704, errno = 1
[0ced17b6490e:00011] Read -1, expected 4704, errno = 1
[0ced17b6490e:00011] Read -1, expected 4224, errno = 1
[0ced17b6490e:00011] Read -1, expected 4224, errno = 1
[0ced17b6490e:00011] Read -1, expected 5632, errno = 1
[0ced17b6490e:00011] Read -1, expected 5632, errno = 1

**Comment by PorousMedia** [comment]
I think the error message is related to the output. Does anyone know what the error means. Also, like @jhuang2601 , if run serially it gives the expected results. Also, if the mpirun is used with a single processor it works fine. So i think its something to do with how the partitions interact. Lastly,  paraview can't open the output file when its ran in parallel, only able to get my observations through the hdf5

**Comment by jhuang2601** [comment]
Hello @TotoGaz,  any idea/suggestion for above mentioned issue with parallel runs? The cases can be running, but outputs cannot be saved correctly.

**Comment by CusiniM** [comment]
@PorousMedia can you create a separate issue for this? I do not think your issue is related to @jhuang2601 's one. We are well aware of some issues with generating conforming fractures in parallel and we are working on a solution but your issues seems different.

**Comment by TotoGaz** [comment]
> I also repeatedly got an error message while it was running.
> [0ced17b6490e:00011] Read -1, expected 4264, errno = 1

Are you running parallel computations in containers (`docker`)?
Check https://github.com/GEOSX/GEOSX/pull/1247

**Comment by jhuang2601** [comment]
Same issue has been reported previously (see [#841](https://github.com/GEOSX/GEOSX/issues/841) )

**Comment by jhuang2601** [comment]
No need for a duplicated report of the same issue (see https://github.com/GEOSX/GEOSX/issues/841 & https://github.com/GEOSX/GEOSX/issues/2014)
